ios/RCTBraintree/RCTBraintree.m

```diff
@@ -218,20 +214,14 @@ - (BTCard*)createCardWithParameters:(NSMutableDictionary*)parameters
         NSString *environment = options[@"environment"];
         NSString *dataSelector = options[@"dataCollector"];
 
-        //Initialize the data collector and specify environment
-        if ([environment isEqualToString:@"development"]) {
-            self.dataCollector = [[BTDataCollector alloc]  initWithEnvironment:BTDataCollectorEnvironmentDevelopment];
-        } else if ([environment isEqualToString:@"qa"]) {
-            self.dataCollector = [[BTDataCollector alloc] initWithEnvironment:BTDataCollectorEnvironmentQA];
-        } else if ([environment isEqualToString:@"sandbox"]) {
-            self.dataCollector = [[BTDataCollector alloc] initWithEnvironment:BTDataCollectorEnvironmentSandbox];
-        }
-
+        //Initialize the data collector in V5
+        self.dataCollector = [[BTDataCollector alloc] initWithAPIClient: self.braintreeClient];
```

Comment:
are development environments for the data collector just not available on Braintree v5?

Reply:
v5 has environments but the method initWithEnvironment is not available in v5. 
 
collectDeviceData() method will now set the environment . Environments are taken directly from the APIClient .
 
BTDataCollectorEnvironment btEnvironment = [self environmentFromString:configuration.environment];
[self setCollectorEnvironment:[self collectorEnvironment:btEnvironment]];

I feel this was an overdo earlier as well , since the method collectDeviceData will eventually call method to set environment .

Reply:
I see, makes sense, thanks for the clarification!

Reply:
Wait so where does the environment get set in your code then? I see where the APIClient gets created but it doesn't look like the environment is being provided to it? Does it just guess based on the client token?

Reply:
So collectDeviceData() inside BrainTree framework will call a function  fetchOrReturnRemoteConfiguration using APIClient. This function will return the configurations for APIClient and the configuration will have the environment. Does this answer your question ?

We have no way of setting the environment  ourselves in v5.